misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ActionNodeVisitor.java

```diff
@@ -203,18 +241,36 @@ private void findInteractions(String methodName, Symbol methodSymbol) {
                             if (node instanceof FunctionDefinitionNode) {
                                 FunctionDefinitionNode functionDefinitionNode = (FunctionDefinitionNode) node;
                                 String referencedFunctionName = functionDefinitionNode.functionName().text();
-                                if (methodName.equals(referencedFunctionName)) {
+                                if ((nameNode instanceof SimpleNameReferenceNode) &&
+                                        ((SimpleNameReferenceNode) nameNode).name().text()
+                                                .equals(referencedFunctionName)) {
+                                    ActionNodeVisitor actionNodeVisitor = new ActionNodeVisitor(packageCompilation,
+                                            nextSemanticModel, currentPackage, this.filePath, visitedFunctionNames);
+                                    functionDefinitionNode.accept(actionNodeVisitor);
+                                    interactionList.addAll(actionNodeVisitor.getInteractionList());
+                                } else if ((nameNode instanceof QualifiedNameReferenceNode) &&
+                                        ((QualifiedNameReferenceNode) nameNode).identifier().text()
+                                                .equals(referencedFunctionName)) {
```

Comment:
Shall we also try to reduce the cognitive complexity of the `findInteractions()` method, as I'm seeing 6-7 nested levels within the function.

We can do so by inverting if conditions with `continue` statements and, using util functions appropriately.